### PR TITLE
Refactored and tested SerializeForLogging

### PR DESCRIPTION
**Pull request by jakobht**
**What changed?**
Refactored SerializeForLogging so we use a wrapper instead of implementing it on every single request type. The function is used to give people implementing custum authorizers the ability to log the request body. 

We supported nil receivers in the old interface, so to not break this support we here support typed nils. 

**Why?**
Improve code quality and test coverage

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/authorization/authorizer_test.go [hunk 1]
```diff
@@ -95,3 +96,89 @@ func Test_validatePermission(t *testing.T) {
...
+		CronSchedule:        "testSchedule",
+		Memo:                &types.Memo{Fields: piiTestMap},
+		SearchAttributes:    &types.SearchAttributes{IndexedFields: piiTestMap},
+		Header:              &types.Header{Fields: map[string][]byte{}},
+		DelayStartSeconds:   &testDelayStartSeconds,
+		JitterStartSeconds:  &testJitterStartSeconds,
+		FirstRunAtTimestamp: &testFirstRunAtTimestamp,
+	}
+	return testReq
+}
```
Comment (jakobht) [inline common/authorization/authorizer_test.go:184]: All this is just moved here from shared_test.go

File: common/authorization/authorizer.go [hunk 2]
```diff
@@ -115,6 +117,27 @@ type FilteredRequestBody interface {
 	SerializeForLogging() (string, error)
 }
 
+type simpleRequestLogWrapper struct {
+	request interface{}
+}
+
+func (f *simpleRequestLogWrapper) SerializeForLogging() (string, error) {
+	if f.request == nil || reflect.ValueOf(f.request).IsNil() {
```
Comment (3vilhamster) [inline common/authorization/authorizer.go:125]: This is a bit inefficient, but this is a minor thing in comparison to the next json.Marshal, so should be fine to use

Comment (jakobht) [reply common/authorization/authorizer.go:128]: Agreed, I added a comment outlining this

**Review by Shaddoll — APPROVED** [review]

**Review by Shaddoll — COMMENTED** [review]
Nice!

**Review by timl3136 — APPROVED** [review]

**Pull request merged**